quantus_sdk/lib/src/services/number_formatting_service.dart

```diff
@@ -67,12 +72,16 @@ class NumberFormattingService {
     }
 
     try {
-      final decimalAmount = Decimal.parse(formattedAmount);
+      final sanitizedText = formattedAmount.replaceAll(',', '.');
+
+      final decimalAmount = Decimal.parse(sanitizedText);
       // Check if input precision exceeds chain precision
       if (decimalAmount.scale > decimals) {
         // Option 1: Truncate (like toBigInt does)
         // Option 2: Throw an error - let's stick with truncation for now
-        debugPrint('Warning: Input amount $formattedAmount exceeds $decimals decimals, will be truncated.');
+        debugPrint(
```

Comment:
we should now allow any entries that are an illegal value. So we only allow up to 12 decimals.